README.md

```diff
@@ -1,25 +1,108 @@
+
+2. Override `FISTrivium`'s default initializer to call the designated initializer. Have it pass in an empty string (`@""`) for the `content` argument, and `0` (zero) for the `likes` argument.
+
+Run the test in `FISTriviumSpec.m` to check the `FISTrivium` class.
+
+### IV. Build the `FISLocation` Class
+
+#### Write the Initializers
+
+1. Define `FISLocation`'s designated initializer.  
+**Hint:** *Don't forget to set the* `trivia` *property to the initialization of an empty mutable array.*
+
+2. Override `FISLocation`'s default initializer to call the designated initializer. Have it pass in an empty string (`@""`) for the `name` argument, and `0.0` for both the `latitude` and `longitude` arguments.
+
+#### Write the Method Implementations
+
+1. Define the `stringByTruncatingNameToLength:` method to return a substring of the location's name shortened to the number of characters defined by the `length` argument.  
+**Hint:** `NSString` *has a handy method called* `substringToIndex`.
+
+2. Define the `validateLocationData` method to:
+  * return `NO` if the `name` property is an empty string or `nil`,
+  * return `NO` if the `latitude` property is beyond `-90.0` and `90.0`,
+  * return `NO` if the `longitude` property is beyond `-180.0` and `180.0` or if it is equal to `-180.0` (since this longitude conceptually matches the 180º meridian), and
+  * otherwise return `YES`.
 
-    ```
-    - (NSString *) content
-    - (NSInteger) likes
-    ```
+3. Define the `triviumWithMostLikes` method to:
+  * return `nil` if the `trivia` array is empty, otherwise
+  * return the `FISTrivium` object in the `trivia` array that has the highest value for the `likes` property.  
+  **Hint:** *You can accomplish this with either a* `for in` *loop &  an* `if` *statement, or by using* `NSSortDescriptor`.
+  
+### V. Write the `FISAppDelegate` Method Implementations
 
-  8. Add an `NSMutableArray` property to `FISLocation` to hold trivia items
-  9. Write a method called `topTrivia` that returns the trivia item with the most likes. Where should this method go?
-  10. Create an initializer for `FISTrivia` that takes an argument for all of the properties and sets them appropriately
+1. Define the `allLocationNames` method to return an array of all of the `name` string properties of the `FISLocation` objects held in the `locations` array property.
 
+2. Define the `locationWithName:` method to return a `FISLocation` object whose `name` string property matches the string submitted to the `name` argument. If there isn't a match, this method should return `nil`.  
+**Hint:** *You can accomplish with either a* `for in` *loop &* `if` *statement, or by using* `NSPredicate`.
 
+3.	Define the `locationsNearLatitude:longitude:margin:` method to return an array of all of the locations whose coordinates are near enough to the submitted coordinates to fall within the submitted positive or negative margin submitted in the `margin` argument.  
```

Comment:
".. all the locations whose coordinates are within ±`margin` degrees from the given location"